### PR TITLE
[71] Implement get jobs for admin API @GET

### PR DESCRIPTION
**Pull request by SuitedIceZ**
## What did you do
- implement ```GET /jobs/admin``` endpoint for admin to get reported jobs and filtering

## Demo
- https://api-dev.modela.miello.dev/swagger#/auth/AuthController_login
- login with (or any admin)
```
{
  "email": "admin1@gmail.com",
  "password": "password"
}
```
- https://api-dev.modela.miello.dev/swagger#/jobs/JobController_findAllByAdmin
- try get jobs with ```REPORTED``` status query on along with any others filters
- should see all reported jobs indicated by boolean ```isReported``` field in each jobs

## Limitation
- not have unit test yet

### CODE REVIEW

File: apps/api/src/modules/job/job.service.ts [hunk 1]
```diff
@@ -16,6 +17,7 @@ import {
   ForbiddenException,
   InternalServerErrorException,
 } from '@nestjs/common/exceptions'
+import console from 'console'
```
Comment (miello) [inline apps/api/src/modules/job/job.service.ts:20]: Is this actually necessary ?

File: apps/api/src/modules/job/job.service.ts [hunk 2]
```diff
@@ -78,14 +80,22 @@ export class JobService {
...
+    const defaultPage = 1
+
+    //set Default value for limit and page
+    searchJobDto.limit = searchJobDto.limit || defaultLimit
+    searchJobDto.page = searchJobDto.page || defaultPage
+
     const params = {
       //take and skip from limit and page
       take: Number(searchJobDto.limit),
       skip: (searchJobDto.page - 1) * searchJobDto.limit,
```
Comment (miello) [inline apps/api/src/modules/job/job.service.ts:93]: Please consider parsing `searchJobDto.page` also

File: packages/dtos/src/job.ts [hunk 3]
```diff
@@ -17,8 +17,9 @@ import { Gender, Job, JobStatus, Shooting } from '@modela/database'
 export enum SearchJobStatus {
   'OPEN' = 'OPEN',
   'CLOSE' = 'CLOSE',
+  'CANCELLED' = 'CANCELLED',
+  'REPORTED' = 'REPORTED',
```
Comment (miello) [inline packages/dtos/src/job.ts:21]: (Discussion)
This is affected to `/jobs` in swagger also 
(Anyone besides Admin should not query with `REPORTED` or `CANCELLED` ?)

image

**Review by miello — CHANGES_REQUESTED** [review]

File: packages/dtos/src/job.ts (hunk 3 above)
Comment (SuitedIceZ) [reply packages/dtos/src/job.ts:21]: hmmm ok, should create separate ENUM for each search type

**Review by miello — APPROVED** [review]

**Pull request merged**